### PR TITLE
Fix duplication around transport code in IndexBasedTransformConfigManager

### PR DESCRIPTION
**Pull request by original-brownbear**
Drying up the logic in this class and making use of
`delegateAndWrap` in an effort to narrow down possible sources for the
remaining very rare leaks we observe around `SearchResponse`.

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java [hunk 1]
```diff
@@ -304,13 +283,13 @@ public void deleteOldIndices(ActionListener<Boolean> listener) {
             IndicesOptions.LENIENT_EXPAND_OPEN
         );
 
-        executeAsyncWithOrigin(client, TRANSFORM_ORIGIN, TransportDeleteIndexAction.TYPE, deleteRequest, ActionListener.wrap(response -> {
+        executeAsyncWithOrigin(TransportDeleteIndexAction.TYPE, deleteRequest, listener.delegateFailureAndWrap((l, response) -> {
```
Comment (przemekwitek) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java:286]: Would it be possible to use either `l` or `delegate` parameter name consistently throughout this file?

Comment (original-brownbear) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/IndexBasedTransformConfigManager.java:286]: Sure thing moved it all to `l` :)

**Review by przemekwitek — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks @przemekwitek !

**Pull request merged**